rust/operator-binary/src/kerberos.rs

```diff
@@ -7,21 +8,25 @@ use stackable_operator::{
     kube::{runtime::reflector::ObjectRef, ResourceExt},
 };
 
-use crate::{
-    config::{CoreSiteConfigBuilder, HdfsSiteConfigBuilder},
-    hdfs_controller::Error,
-};
+use crate::config::{CoreSiteConfigBuilder, HdfsSiteConfigBuilder};
 
-pub fn check_if_supported(resolved_product_image: &ResolvedProductImage) -> Result<(), Error> {
-    // We only support Kerberos for HDFS >= 3.3.x
-    // With HDFS 3.2.2 we got weird errors, which *might* be caused by DNS lookup issues
-    // The Stacktrace is documented in rust/operator/src/kerberos_hdfs_3.2_stacktrace.txt
+type Result<T, E = Error> = std::result::Result<T, E>;
 
-    if resolved_product_image.product_version.starts_with("3.2.") {
-        Err(Error::KerberosNotSupported {})
-    } else {
-        Ok(())
-    }
+#[derive(Snafu, Debug)]
+#[allow(clippy::enum_variant_names)]
+pub enum Error {
+    #[snafu(display("Object has no namespace"))]
+    ObjectHasNoNamespace {
+        source: stackable_hdfs_crd::Error,
+        obj_ref: ObjectRef<HdfsCluster>,
+    },
+}
+
+/// We only support Kerberos for HDFS >= 3.3.x
+/// With HDFS 3.2.2 we got weird errors, which *might* be caused by DNS lookup issues
+/// The Stacktrace is documented in rust/operator/src/kerberos_hdfs_3.2_stacktrace.txt
+pub fn check_if_supported(resolved_product_image: &ResolvedProductImage) -> bool {
```

Comment:
Alternatively..

```suggestion
#[derive(Debug, Snafu)]
struct KerberosNotSupported;
pub fn require_support(resolved_product_image: &ResolvedProductImage) -> Result<(), KerberosNotSupported> {
```

Reply:
I turned it into a bool to make the body simpler, and so that we didn't need duplicate Snafus up the call chain.

Reply:
Renamed in 8b25f33ab422c3fd908dc1de226e6896f185e4d3

Reply:
Ohhh good point, I missed that part.

Resolved in e7160ad97d7beb9d96e057a07b3e0d87e993805e

Reply:
There's also https://github.com/stackabletech/hdfs-operator/pull/454/files#diff-a4441abfc6a76cee374b7e2b80a8a76901cbff6d19877be5b13de19453d4604aR246.

Reply:
Ahh :facepalm:
Maybe I should just name it `is_not_supported`